src/context/mod.rs

```diff
@@ -207,7 +206,47 @@ impl<'ctx> WorkflowContext<'ctx> {
     }
 }
 
-/// Trait exposing Restate timers functionalities.
+///
+/// # Scheduling & Timers
+/// The Restate SDK includes durable timers.
+/// You can use these to let handlers sleep for a specified time, or to schedule a handler to be called at a later time.
+/// These timers are resilient to failures and restarts.
+/// Restate stores and keeps track of the timers and triggers them on time, even across failures and restarts.
+///
+/// ## Scheduling Async Tasks
+///
+/// To schedule a handler to be called at a later time, have a look at the documentation on [delayed calls][crate::context::ContextClient#delayed-calls].
+///
+///
+/// ## Durable sleep
+/// To sleep in a Restate application for ten seconds, do the following:
+///
+/// ```rust,no_run
+/// # use restate_sdk::prelude::*;
+/// # use std::convert::Infallible;
+/// # use std::time::Duration;
+/// #
+/// # async fn handle(ctx: Context<'_>) -> Result<(), HandlerError> {
+/// ctx.sleep(Duration::from_secs(10)).await?;
+/// #    Ok(())
+/// # }
+/// ```
+///
+/// **Tip: Cost savings on FaaS**:
+/// Restate suspends the handler while it is sleeping, to free up resources.
+/// This is beneficial for AWS Lambda deployments, since you don't pay for the time the handler is sleeping.
+///
+/// **Info: Sleeping in Virtual Objects**:
+/// Virtual Objects only process a single invocation at a time, so the Virtual Object will be blocked while sleeping.
```

Comment:
This belongs to the generic description about properties of virtual objects i mentioned in another comment

Reply:
I'll leave it here as a note.